utils.py

```diff
@@ -40,3 +44,180 @@ def to_snake(name):
     Convert plugin name from snake case to dash representation
     """
     return name.replace("-", "_")
+
+
+# Changelog to markdown migration
+
+
+def migrate_changelog_to_markdown(plugin_root: str):
+    """
+    - Converts the changelog if there is a CHANGES.rst
+    - Updates the pyproject.toml "tool.towncrier" section to use markdown if it doesnt.
+    """
+    root_path = Path(plugin_root)
+    plugin_name = root_path.stem
+
+    # pyproject update
+    pyproject_file = root_path / "pyproject.toml"
+    if (
+        r'start_string = "[//]: # (towncrier release notes start)\n"'
+        not in pyproject_file.read_text()
+    ):
+        update_pyproject_toml(pyproject_file)
+
+    # changelog conversion
+    rst_changes = root_path / "CHANGES.rst"
+    if rst_changes.exists():
+        convert_changelog(rst_changes, plugin_name)
+        rst_changes.unlink()
+
+
+def update_pyproject_toml(pyproject_file: Path) -> Path:
+    import tomlkit
+
+    pyproject_data = tomlkit.loads(pyproject_file.read_text())
+
+    package_name = pyproject_data["tool"]["towncrier"]["package"]  # type: ignore
+    TOWNCRIER_TOML_DATA = {
+        "package": package_name,
+        "filename": "CHANGES.md",
+        "directory": "CHANGES/",
+        "title_format": "## {version} ({project_date}) {{: #{version} }}",
+        "template": "CHANGES/.TEMPLATE.md",
+        "issue_format": Template(
+            "[#{issue}](https://github.com/pulp/$plugin_name/issues/{issue})"
+        ).substitute(plugin_name=package_name),
```

Comment:
This will not always give the right result.
Can we hand the context for the jinja2 templates down here? There should be a repository variable in it.

Reply:
I believe we can't hand this to jinja (for example, add a `{{ repository_name }}` )because on the CI update the plugin-template won't touch this file (pyproject.toml), as far as I can tell.

Reply:
What about using the plugin name derived from the repository path (as in line 58)? That should strictly match the repository name.

Reply:
Whatever gives us the proper "repository_path" should be fine. But pulp repositories _can_ absolutely live outside of the pulp org.